excel-importer/src/main/java/uk/gov/hmcts/ccd/definition/store/excel/service/ImportServiceImpl.java

```diff
@@ -150,9 +154,19 @@ public DefinitionFileUploadMetadata importFormDefinitions(InputStream inputStrea
         logger.debug("Importing spreadsheet: Case types...");
 
         final CaseTypeParser caseTypeParser = parserFactory.createCaseTypeParser(parseContext);
-        final ParseResult<CaseTypeEntity> parsedCaseTypes = caseTypeParser.parseAll(definitionSheets);
+        Set<String> missingUserRoles = Sets.newHashSet();
+        final ParseResult<CaseTypeEntity> parsedCaseTypes = caseTypeParser.parseAll(definitionSheets, missingUserRoles);
         List<CaseTypeEntity> caseTypes = parsedCaseTypes.getNewResults();
-        caseTypeService.createAll(jurisdiction, caseTypes); // runs validation
+        try {
+            caseTypeService.createAll(jurisdiction, caseTypes); // runs validation
+        } catch (ValidationException ve) {
+            checkMissingUserRoles(missingUserRoles);
+            throw ve;
+        }
+
+        checkMissingUserRoles(missingUserRoles);
```

Comment:
can't we have this check moved into the caseTypeParser itself? so that the case type parser throws the error if it detects there are missing user roles

Reply:
We can do that, but it will not log the error messages, which is happening in 
caseTypeService.createAll(jurisdiction, caseTypes);